controlled_resource_handler.go

```diff
@@ -9,107 +9,105 @@ import (
 			return
 		}
-		logger = g.loggerForObj(metaObj)
 	}
+	logger := g.loggerForObj(metaObj)
 	name, namespace := g.getControllerNameAndNamespace(metaObj)
 	if name == "" { // No controller object found
-		if g.CreatorIndex != nil {
-			creators, err := g.CreatorIndex.CreatorByObject(
+		if g.ControllerIndex != nil {
+			controllers, err := g.ControllerIndex.ControllerByObject(
 				metaObj.(runtime.Object).GetObjectKind().GroupVersionKind().GroupKind(), namespace, metaObj.GetName())
 			if err != nil {
-				logger.Error("Failed to get creators for object", zap.Error(err))
+				logger.Error("Failed to get controllers for object", zap.Error(err))
 				return
 			}
-			for _, creator := range creators {
-				g.rebuildByName(logger, namespace, creator.(meta_v1.Object).GetName(), "deleted")
+			for _, controller := range controllers {
+				g.rebuildControllerByName(logger, namespace, controller.(meta_v1.Object).GetName(), "deleted")
 			}
 		}
 	} else {
-		g.rebuildByName(logger, namespace, name, "deleted")
+		g.rebuildControllerByName(logger, namespace, name, "deleted")
 	}
 }
 
-// This method may be called with an empty name.
-func (g *ControlledResourceHandler) rebuildByName(logger *zap.Logger, namespace, name, addUpdateDelete string) {
-	if name == "" {
+// This method may be called with an empty controllerName.
+func (g *ControlledResourceHandler) rebuildControllerByName(logger *zap.Logger, namespace, controllerName, addUpdateDelete string) {
+	if controllerName == "" {
 		return
 	}
 	logger.
-		With(g.ZapNameField(name)).
-		Sugar().Infof("Rebuilding creator object %q because created object was %s", name, addUpdateDelete)
+		With(logz.ControllerName(controllerName)).
+		Sugar().Infof("Enqueuing controller object because controlled object was %s", addUpdateDelete)
 	g.WorkQueue.Add(QueueKey{
 		Namespace: namespace,
-		Name:      name,
+		Name:      controllerName,
 	})
 }
 
 // getControllerNameAndNamespace returns name and namespace of the object's controller.
 // Returned name may be empty if the object does not have a controller owner reference.
-func (g *ControlledResourceHandler) getControllerNameAndNamespace(obj interface{}) (string, string) {
+func (g *ControlledResourceHandler) getControllerNameAndNamespace(obj meta_v1.Object) (string, string) {
 	var name string
-	meta := obj.(meta_v1.Object)
-	ref := meta_v1.GetControllerOf(meta)
+	ref := meta_v1.GetControllerOf(obj)
 	if ref != nil && ref.APIVersion == g.ControllerGvk.GroupVersion().String() && ref.Kind == g.ControllerGvk.Kind {
 		name = ref.Name
 	}
-	return name, meta.GetNamespace()
+	return name, obj.GetNamespace()
 }
 
-func (g *ControlledResourceHandler) loggerForObj(obj interface{}) *zap.Logger {
-	logger := g.Logger
-	metaObj, ok := obj.(meta_v1.Object)
-	if ok { // This is conditional to deal with tombstones on delete event
-		logger = logger.With(logz.Namespace(metaObj), g.ZapNameField(metaObj.GetName()))
```

Comment:
This is the bug - it is using the name function for a controller with a name of a controlled object. Mess in the logs.